src/wrappers/themis/rust/src/error.rs

```diff
@@ -203,3 +215,62 @@ pub enum ErrorKind {
     /// Attempt to use Secure Comparator before completing nonce exchange.
     CompareNotReady,
 }
+
+// TransportError does not implement PartialEq therefore it's not possible to derive the
+// implementation for ErrorKind automatically. TransportErrors cannot be compared because
+// they either contain arbitrary human-readable strings or abstract boxed errors which
+// may have different types. However, we really need to be able to compare ErrorKinds.
+// Implement comparison by ignoring differences in TransportError details, comparing
+// just *kinds* of errors, not their values. This is a full equivalence relationship.
+
+impl Eq for ErrorKind {}
+
+impl PartialEq for ErrorKind {
+    fn eq(&self, other: &ErrorKind) -> bool {
+        error_kinds_equal(self, other)
+    }
+}
+
+impl PartialEq<&ErrorKind> for ErrorKind {
+    fn eq(&self, other: &&ErrorKind) -> bool {
+        error_kinds_equal(self, other)
+    }
+}
+
+impl PartialEq<ErrorKind> for &ErrorKind {
+    fn eq(&self, other: &ErrorKind) -> bool {
+        error_kinds_equal(self, other)
+    }
+}
+
+fn error_kinds_equal(lhs: &ErrorKind, rhs: &ErrorKind) -> bool {
```

Comment:
will be good to add a comment to ErrorKind definition that we should update `error_kinds_equal` if some new values will be added

Reply:
Yeah... That's unfortunate but the catch-all branch in this match makes it possible to add a new variant to `ErrorKind` without adding a corresponding line to the comparison implementation.

I guess there's no other way around, I'll add a comment and let's hope it will be noticed.